tests/integration/test_deduplication.py

```diff
@@ -57,3 +57,20 @@ async def test_multiple_scrape_jobs_in_constructor(
     targets = await get_prometheus_active_targets(ops_test, prometheus_app_name)
     # Two unique jobs above plus an additional an additional job for self scraping.
     assert len(targets) == 3
+
+
+async def test_same_app_related_two_ways(
+    ops_test: OpsTest, prometheus_charm, prometheus_tester_charm
+):
+    """Test that the deduplication works when the same app is related twice."""
+    await asyncio.gather(
+        ops_test.model.applications[tester_app_name].set_config({"scrape_jobs": "[]"}),
```

Comment:
Why clear the config?

Reply:
I want to make sure they both have the default job so they are the same.

Reply:
In that case I you may want to use `.reset_config(["scrape_jobs"])` in case we change the default value in the future.

Reply:
Good call. I didn't notice that method.

Reply:
Done.